llvm/lib/CodeGen/MachineVerifier.cpp

```diff
@@ -1435,7 +1435,7 @@ void MachineVerifier::verifyPreISelGenericInstruction(const MachineInstr *MI) {
     if (DstTy.getElementType() != SrcEltTy)
       report("G_BUILD_VECTOR result element type must match source type", MI);
 
-    if (DstTy.getNumElements() != MI->getNumOperands() - 1)
+    if (DstTy.getElementCount().getKnownMinValue() > MI->getNumOperands() - 1)
       report("G_BUILD_VECTOR must have an operand for each elemement", MI);
```

Comment:
How can a G_BUILD_VECTOR work for scalable vectors. The number of scalars needed isn't known at compile time.

Reply:
I will update this. I guess it is the case that G_BUILD_VECTOR cannot work for scalable vectors unless both Src and Dest types are scalable vectors and `N * T1 * NumDests = M * T2`? But that is a story for a different PR. I will revert this change here.

Reply:
I think that would be a G_CONCAT_VECTORS?

Reply:
Ya thats a good point.